enumfields/fields.py

```diff
@@ -179,26 +215,24 @@ def _get_all_parent_choices(self, supvalue):
             choice for choice in self.enum if supvalue in choice.parents
         }
 
-    def clean(self, value, model_instance):
-        if self._get_all_parent_choices(self._get_supvalue(model_instance)):
-            return super().clean(value, model_instance)
-        else:
-            return None
-
-    def _raise_error_if_value_should_be_empty(self, value, supvalue):
+    def _raise_error_if_parent_value_should_be_empty(self, value, supvalue):
         if supvalue not in self._get_all_parent_values() and value is not None:
             raise ValidationError(ugettext('Value must be empty'))
 
-    def _raise_error_if_value_not_allowed(self, value, supvalue):
+    def _raise_error_if_parent_value_not_allowed(self, value, supvalue):
         allowed_values = self._get_all_parent_choices(supvalue)
         if allowed_values and value not in allowed_values:
             raise ValidationError(ugettext('Allowed choices are {}.').format(
                 ', '.join(('{} ({})'.format(*(val.label, val)) for val in allowed_values))
             ))
 
     def validate(self, value, model_instance):
-        self._raise_error_if_value_should_be_empty(value, self._get_supvalue(model_instance))
-        self._raise_error_if_value_not_allowed(value, self._get_supvalue(model_instance))
+        parent_value = self._get_supvalue(model_instance)
```

Comment:
I am not getting what is parent value?

Reply:
This is a subfield. That's the value of the parent field. I renamed it to `parent_field_value`. Anyway this code was already there.